openupgrade_scripts/scripts/sale_stock/16.0.1.0/upgrade_analysis_work.txt

```diff
@@ -0,0 +1,15 @@
+---Models in module 'sale_stock'---
+---Fields in module 'sale_stock'---
+sale_stock   / sale.order               / delivery_status (selection)   : NEW selection_keys: ['full', 'partial', 'pending'], isfunction: function, stored
```

Comment:
@huguesdk I think it's necessary to precompute because the number of records in sale.order will be large, and its logic is also quite simple. I have implemented it here https://github.com/OCA/OpenUpgrade/pull/4057 for you to consider, and one of the two PRs might be closed.

Reply:
thanks @hoangtiendung070797 for answering my question and providing a solution. i added the precomputation using a different query which should be more efficient. i also used `openupgrade.add_fields()` instead of altering the table directly, as is done in [the documentation](https://oca.github.io/OpenUpgrade/use_cases/field_fast_precomputation.html). what do you think?

happy 26th birthday! (guessed from your username) 😉

Reply:
@huguesdk Yes, today is my 26th birthday. Thank you very much :smile:

Reply:
@huguesdk your query is very short, i will try its effectiveness and close my PR. Thank you